impy/common.py

```diff
@@ -535,6 +626,32 @@ def _define_default_fs_particles(self):
         if impy_config['pi0_stable']:
             self.set_stable(111)
 
+    def __call__(self, nevents):
+        """Generator function (in python sence) 
+        which launches the underlying event generator 
+        and returns its the result (event) as MCEvent object
+        """
+        retry_on_rejection = impy_config['retry_on_rejection']
+        # Initialize counters to prevent infinite loops in rejections
+        ntrials = 0
+        nremaining = nevents
+        self.nevents = nevents
+        self.norm = 1./float(nevents)
+        while nremaining > 0:
+            if self.generate_event() == 0:
+                yield self._event_class(self.lib, self._curr_event_kin,
+                                        self._output_frame)
+                nremaining -= 1
+                ntrials += 1
+            elif retry_on_rejection:
+                info(10, 'Rejection occured. Retrying..')
```

Comment:
I haven't seen yet how `info` is implemented, but if this a custom solution, I recommend to use one of the major logging libraries, because doing logging right is not trivial.

Reply:
I cannot answer much to this comment. Because the `info` function is scattered around the code it is not so easy to change logging system now. Moreover, one should decide what to use instead. I think it deserves its own issue.

Reply:
You are right, for this PR there is no change needed.

Reply:
Info is implemented in util and prints out the caller name (class and method) along with the message, if the level is above the one defined by config['debug_level'], otherwise it just returns without.

Not sure we really need a super extensive logging, but if, it should be a drop in replacement anyways, and it should print the caller name. Also it hopefully is a light dependency without involving too many crazy requirements.